compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -15157,7 +15157,8 @@ private STNode parseTypedBindingPatternOrExprRhs(STNode typeOrExpr, boolean allo
                 return STNodeFactory.createBinaryExpressionNode(SyntaxKind.BINARY_EXPRESSION, typeOrExpr,
                         pipeOrAndToken, rhsTypedBPOrExpr);
             case SEMICOLON_TOKEN:
-                if (isDefiniteExpr(typeOrExpr.kind)) {
+                if (typeOrExpr.kind.compareTo(SyntaxKind.BINARY_EXPRESSION) >= 0
+                        && typeOrExpr.kind.compareTo(SyntaxKind.ERROR_CONSTRUCTOR) <= 0) {
```

Comment:
same condition is repeated inside `isDefiniteExpr()` as well. Therefore, shall we extract this to a method say `isExpression()`?

Reply:
Updated 95cca811288f371b680fde4962089b1eb94ab90c